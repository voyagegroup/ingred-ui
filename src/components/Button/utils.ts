import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
