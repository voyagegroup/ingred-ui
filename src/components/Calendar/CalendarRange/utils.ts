import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
