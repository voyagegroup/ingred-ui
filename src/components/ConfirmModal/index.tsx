export { default } from "./ConfirmModal";
export type {
  ConfirmModalProps,
  ConfirmModalCloseReason,
} from "./ConfirmModal";
