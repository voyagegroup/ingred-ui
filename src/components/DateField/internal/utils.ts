export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
