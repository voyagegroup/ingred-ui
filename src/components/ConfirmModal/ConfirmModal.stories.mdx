import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import ConfirmModal from "./ConfirmModal";
import Button from "../Button";
import DataTable from "../DataTable";
import { data } from "../DataTable/mockData";

<Meta
  title="Components/Utils/ConfirmModal"
  component={ConfirmModal}
  parameters={{ docs: { source: { type: "code" } } }}
  argTypes={{ onSubmit: { action: "submitted" } }}
/>

# ConfirmModal

<ArgsTable of={ConfirmModal} />

## Samples

### Basic

<Canvas>
  <Story
    name="basic"
    args={{
      isOpen: false,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            onClose={handleToggleButton}
            {...args}
            isOpen={isOpen}
          >
            Content
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>

### Full Size

<Canvas>
  <Story
    name="full size"
    args={{
      isOpen: false,
      fullSize: true,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            fullSize={true}
            onClose={handleToggleButton}
            {...args}
            isOpen={isOpen}
          >
            Content
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>

### With tips

<Canvas>
  <Story
    name="with tips"
    args={{
      isOpen: false,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            tipElement={<div>Description for this modal.</div>}
            onClose={handleToggleButton}
            {...args}
            isOpen={isOpen}
          >
            Content
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>

### With subactions

<Canvas>
  <Story
    name="with subactions"
    args={{
      isOpen: false,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            onClose={handleToggleButton}
            {...args}
            fullSize={true}
            subActions={[
              {
                title: "Download in CSV format",
                icon: "export",
                action: () => {},
              },
              {
                title: "Add to dashboard",
                icon: "export",
                action: () => {},
              },
            ]}
            isOpen={isOpen}
          >
            Content
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>

### Loading

<Canvas>
  <Story
    name="loading"
    args={{
      isOpen: false,
      loading: true,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            loading={true}
            onClose={handleToggleButton}
            {...args}
            isOpen={isOpen}
          >
            Content
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>

### Overflow Y scroll

<Canvas>
  <Story
    name="overflow Y scroll"
    args={{
      isOpen: false,
      overflowYScroll: true,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            overflowYScroll={true}
            onClose={handleToggleButton}
            {...args}
            isOpen={isOpen}
          >
            <DataTable
              data={data}
              columns={[
                {
                  name: "ID",
                  selector: (data) => data.id,
                },
                {
                  name: "name",
                  selector: (data) => data.date,
                },
              ]}
            />
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>

### Without footer

<Canvas>
  <Story
    name="without footer"
    args={{
      isOpen: false,
      onSubmit: undefined,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleButton = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleButton}>Open Modal</Button>
          <ConfirmModal
            title="Title"
            onClose={handleToggleButton}
            onSubmit={undefined}
            {...args}
            isOpen={isOpen}
          >
            Content
          </ConfirmModal>
        </>
      );
    }}
  </Story>
</Canvas>
