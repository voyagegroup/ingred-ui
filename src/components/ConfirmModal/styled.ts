import styled, { keyframes, css } from "styled-components";
import { addScrollbarProperties } from "../../utils/scrollbar";
import Card from "../Card";

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;
export const ModalBackground = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  /* TODO: color */
  background-color: #001326cb;
  animation: ${fadeIn} 0.4s;
`;

const slideIn = keyframes`
  0% {
    transform: translate(-50%, calc(-50% + 8px));
  }
  100% {
    transform: translate(-50%, -50%);
  }
`;

const FullSizeSlideIn = keyframes`
  0% {
    transform: translate(-50%, 0%);
  }
  100% {
    transform: translate(-50%, -48%);
  }
`;
export const ModalContainer = styled(Card)<{ fullSize: boolean }>`
  position: absolute;
  top: 50%;
  left: 50%;
  min-width: 400px;
  width: ${({ fullSize }) => (fullSize ? "100vw" : "auto")};
  height: ${({ fullSize }) => (fullSize ? "100vh" : "auto")};
  transform: ${({ fullSize }) =>
    fullSize ? `translate(-50%, -48%);` : `translate(-50%, -50%);`};
  border-radius: ${({ fullSize, theme }) =>
    fullSize ? `8px 8px 0 0` : `${theme.radius}px`};
  background-color: ${({ theme }) => theme.palette.background.default};
  animation: ${({ fullSize }) => (fullSize ? FullSizeSlideIn : slideIn)} 0.4s;
  overflow: hidden;
`;

export const ModalHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${({ theme }) => theme.spacing * 2}px
    ${({ theme }) => theme.spacing * 3}px;
  border-bottom: 1px solid ${({ theme }) => theme.palette.gray.light};
`;

export const LeftContainer = styled.div`
  display: flex;
  align-items: center;
  max-width: 70%;
`;

type ScrollContainerProps = {
  overflowYScroll: boolean;
  fullSize: boolean;
  showFooter: boolean;
};

export const ScrollContainer = styled.div<ScrollContainerProps>`
  height: ${({ fullSize, showFooter, theme }) =>
    fullSize
      ? `calc(100vh - ${theme.spacing * 2 * 2 + 42 + (showFooter ? 61 : 0)}px)`
      : "auto"};
  margin-bottom: ${({ fullSize, showFooter, theme }) =>
    /* MEMO: Height of ModalFooter(padding-top + padding-bottom + Button size="medium") */
    fullSize || !showFooter ? 0 : theme.spacing * 2 * 2 + 42}px;
  ${({ overflowYScroll, fullSize }) =>
    overflowYScroll
      ? addScrollbarProperties({
          maxHeight: fullSize ? "auto" : "calc(80vh - 61px)",
        })
      : css`
          overflow-y: visible;
          max-height: ${fullSize ? "auto" : "calc(80vh - 61px)"};
        `}
`;

export const ModalFooter = styled.div<{ fullSize: boolean }>`
  position: fixed;
  left: 0;
  bottom: 0;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  padding: ${({ theme }) => theme.spacing * 2}px
    ${({ theme }) => theme.spacing * 3}px;
  background-color: ${({ theme }) => theme.palette.gray.highlight};
  border-radius: ${({ fullSize, theme }) =>
    fullSize ? 0 : `0 0 ${theme.radius}px ${theme.radius}px`};
  margin-bottom: ${({ fullSize }) => (fullSize ? "1.8vh" : "auto")};
`;

export const IconContainer = styled.div`
  cursor: pointer;
`;

export const LoadingContainer = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(255, 255, 255, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const TipContainer = styled.div`
  padding-left: ${({ theme }) => theme.spacing * 0.5}px;
`;

export const TipContentContainer = styled.div`
  min-width: ${({ theme }) =>
    360 - (theme.spacing * 2 * 2 + theme.spacing * 2 + 18)}px;
  max-width: ${({ theme }) =>
    512 - (theme.spacing * 2 * 2 + theme.spacing * 2 + 18)}px;
`;
