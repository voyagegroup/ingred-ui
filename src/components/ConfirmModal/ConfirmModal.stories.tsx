import React from "react";
import { StoryObj } from "@storybook/react";
import { Title, ArgsTable, Stories } from "@storybook/addon-docs";
import ConfirmModal, { ConfirmModalProps } from "./ConfirmModal";
import { action } from "@storybook/addon-actions";
import Button from "../Button";
import DataTable from "../DataTable";
import { data } from "../DataTable/mockData";
import Spacer from "../Spacer";
import ActionButton from "../ActionButton";

export default {
  title: "Components/Utils/ConfirmModal",
  component: ConfirmModal,
  args: {
    isOpen: false,
    title: "Title",
  },
  parameters: {
    docs: {
      source: { language: "tsx" },
      page: () => (
        <>
          <Title />
          <ArgsTable of={ConfirmModal} />
          <Stories includePrimary title="Stories" />
        </>
      ),
    },
  },
};

const Template: StoryObj<ConfirmModalProps> = {
  render: (args) => {
    const [isOpen, setIsOpen] = React.useState(args.isOpen);
    const handleToggleButton = () => {
      setIsOpen(!isOpen);
    };
    return (
      <>
        <Button onClick={handleToggleButton}>Open Modal</Button>
        <ConfirmModal onClose={handleToggleButton} {...args} isOpen={isOpen}>
          {args.children}
        </ConfirmModal>
      </>
    );
  },
};

export const Basic = {
  ...Template,
  args: {
    children: "Content",
    onSubmit: action("submitted"),
  },
};

export const WithOneSubAction = {
  ...Template,
  args: {
    children: "Content",
    onSubmit: action("submitted"),
    subActions: [
      <ActionButton
        icon="export"
        type="button"
        onClick={action(`clicked "Add"`)}
      >
        Add
      </ActionButton>,
    ],
  },
};

export const WithTwoSubActions = {
  ...Template,
  children: "Content",
  onSubmit: action("submitted"),
  subActions: [
    <ActionButton icon="export" type="button" onClick={action(`clicked "Add"`)}>
      Add
    </ActionButton>,
    <ActionButton
      icon="export"
      type="button"
      onClick={action(`clicked "Download"`)}
    >
      Download
    </ActionButton>,
  ],
};

export const Loading = {
  ...Template,
  args: { children: "Content", onSubmit: () => {}, loading: true },
};

export const OverflowYScroll = {
  ...Template,
  args: {
    children: (
      <>
        <Spacer my={3} />
        <DataTable
          data={data}
          dataKey="id"
          columns={[
            {
              name: "ID",
              selector: (data) => data.id,
            },
            {
              name: "name",
              selector: (data) => data.date,
            },
          ]}
        />
      </>
    ),
    onSubmit: action("submitted"),
    overflowYScroll: true,
  },
};

export const WithoutFooter = {
  ...Template,
  args: { children: "Content", onSubmit: undefined },
};
