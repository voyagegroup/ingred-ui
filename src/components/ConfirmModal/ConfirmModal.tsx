import * as React from "react";
import * as Styled from "./styled";
import Icon from "../Icon";
import Spacer from "../Spacer";
import Flex from "../Flex";
import Button from "../Button";
import { ButtonColor } from "../Button/Button";
import Spinner from "../Spinner";
import { useTheme } from "../../themes";
import { IconName } from "../Icon/Icon";
import Modal, { ModalCloseReason, ModalProps } from "../Modal";
import Fade from "../Fade";
import { CSSTransitionProps } from "../../utils/reactTransitionGroup";
import { useLocaleProps } from "../../hooks/useLocaleProps";
import Typography from "../Typography";

export type ConfirmModalCloseReason = "clickCloseIcon" | "clickCancelButton";

export type SubAction = {
  title: string;
  icon: IconName;
  action: () => void;
};

export type ConfirmModalProps = {
  title: string;
  confirmText?: string;
  cancelText?: string;
  onClose?: (
    event:
      | React.MouseEvent<HTMLButtonElement, MouseEvent>
      | React.MouseEvent<Element, MouseEvent>,
    reason: ModalCloseReason | ConfirmModalCloseReason,
  ) => void;
  /**
   * If `undefined`, footer is removed.
   */
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  buttonColor?: ButtonColor;
  isOpen?: boolean;
  /**
   * If `true`, disabled to click submit button.
   */
  disabled?: boolean;
  loading?: boolean;
  overflowYScroll?: boolean;
  /**
   * If `true`, children is wrapped by `<Spacer px={3} pb={3} />`.
   */
  disableHorizontalPadding?: boolean;
  subActions?: React.ReactNode[];
  /**
   * props of [Modal](/?path=/docs/components-utils-modal)
   */
  modalProps?: ModalProps;
  /**
   * props of [Fade](/?path=/docs/components-utils-fade)
   */
  fadeProps?: CSSTransitionProps;

  // MEMO: Added this to prevent type error.
  children?: React.ReactNode;
};

const ConfirmModal = React.forwardRef<HTMLDivElement, ConfirmModalProps>(
  (inProps, ref) => {
    const props = useLocaleProps({ props: inProps, name: "ConfirmModal" });
    const {
      title,
      confirmText = "Confirm",
      cancelText = "Cancel",
      children,
      onClose,
      onSubmit,
      buttonColor = "primary",
      isOpen = true,
      disabled,
      loading,
      overflowYScroll = true,
      disableHorizontalPadding = false,
      subActions,
      modalProps,
      fadeProps,
    } = props;

    const theme = useTheme();
    const showFooter = !!onSubmit;

    const handleClose =
      (reason: ConfirmModalCloseReason) =>
      (
        event:
          | React.MouseEvent<HTMLButtonElement, MouseEvent>
          | React.MouseEvent<Element, MouseEvent>,
      ) => {
        if (onClose) onClose(event, reason);
      };

    return (
      <Modal isOpen={isOpen} onClose={onClose} {...modalProps}>
        <Fade in={isOpen} unmountOnExit={true} {...fadeProps}>
          <Styled.ModalContainer ref={ref}>
            <form onSubmit={onSubmit}>
              <Styled.ModalHeader>
                <Styled.LeftContainer>
                  <Typography weight="bold" size="xxxl">
                    {title}
                  </Typography>
                  {subActions && (
                    <Spacer pl={2}>
                      <Flex display="flex" alignItems="center" gap={1}>
                        {subActions.map((action, index) => (
                          // eslint-disable-next-line react/no-array-index-key
                          <React.Fragment key={index}>{action}</React.Fragment>
                        ))}
                      </Flex>
                    </Spacer>
                  )}
                </Styled.LeftContainer>
                <Styled.IconContainer onClick={handleClose("clickCloseIcon")}>
                  <Icon name="close" size="lg" color={theme.palette.black} />
                </Styled.IconContainer>
              </Styled.ModalHeader>
              <Styled.ScrollContainer
                overflowYScroll={overflowYScroll}
                showFooter={showFooter}
              >
                {disableHorizontalPadding ? (
                  children
                ) : (
                  <Spacer px={3} pb={3}>
                    {children}
                  </Spacer>
                )}
              </Styled.ScrollContainer>
              {showFooter && (
                <Styled.ModalFooter>
                  <Flex display="flex" alignItems="center">
                    <Button
                      type="button"
                      color="secondary"
                      inline={true}
                      onClick={handleClose("clickCancelButton")}
                    >
                      {cancelText}
                    </Button>
                    <Spacer pr={2} />
                    <Button
                      type="submit"
                      color={buttonColor}
                      disabled={disabled}
                      inline={true}
                    >
                      {confirmText}
                    </Button>
                  </Flex>
                </Styled.ModalFooter>
              )}
            </form>
            {loading && (
              <Styled.LoadingContainer>
                <Spinner />
              </Styled.LoadingContainer>
            )}
          </Styled.ModalContainer>
        </Fade>
      </Modal>
    );
  },
);

export default ConfirmModal;
