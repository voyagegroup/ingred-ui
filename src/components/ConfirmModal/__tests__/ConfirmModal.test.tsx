import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import ConfirmModal from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

jest.mock("react-dom", () => {
  const original = jest.requireActual("react-dom");
  return {
    ...original,
    createPortal: (node: any) => node,
  };
});

describe("ConfirmModal component testing", () => {
  afterEach(cleanup);

  test("ConfirmModal", () => {
    const { asFragment } = renderWithThemeProvider(
      <ConfirmModal title="Title" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
