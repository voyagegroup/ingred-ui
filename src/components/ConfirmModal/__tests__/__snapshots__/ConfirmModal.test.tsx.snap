// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal component testing ConfirmModal 1`] = `
<DocumentFragment>
  <div
    class="sc-fFeiMQ fJuGfT"
  >
    <div
      class="sc-ieecCq ixODts fade-transition-appear fade-transition-appear-active"
    >
      <div
        class="sc-bkkeKt fEwjBL"
      />
    </div>
    <div
      class="sc-ieecCq ixODts fade-transition-appear fade-transition-appear-active"
    >
      <div
        class="sc-gsDKAQ sc-hKwDye gmmrAP cnIyXF"
      >
        <form>
          <div
            class="sc-eCImPb hMUlJt"
          >
            <div
              class="sc-jRQBWg gNzDsc"
            >
              <p
                class="sc-iqseJM bkdyck"
                color="#041C33"
                font-size="20px"
              >
                Title
              </p>
              <div
                class="sc-egiyK haNrTA"
              />
            </div>
            <div
              class="sc-furwcr enHelU"
            >
              <div
                class="sc-crHmcD hehbzV"
                size="24"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#041C33"
                    height="23.141"
                    rx="2.571"
                    transform="translate(18.364 2) rotate(45)"
                    width="5.142"
                  />
                  <rect
                    fill="#041C33"
                    height="23.141"
                    rx="2.571"
                    transform="translate(2 5.636) rotate(-45)"
                    width="5.143"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="sc-gKclnd dmgRET"
          >
            <div
              class="sc-egiyK eHvOtZ"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
